packages/core/test/lib/utils/applyScopeDataToEvent.test.ts

```diff
@@ -158,3 +164,72 @@ describe('mergeScopeData', () => {
     });
   });
 });
+
+describe('applyScopeDataToEvent', () => {
+  it("doesn't apply scope.transactionName to transaction events", () => {
+    const data: ScopeData = {
+      eventProcessors: [],
+      breadcrumbs: [],
+      user: {},
+      tags: {},
+      extra: {},
+      contexts: {},
+      attachments: [],
+      propagationContext: { spanId: '1', traceId: '1' },
+      sdkProcessingMetadata: {},
+      fingerprint: [],
+      transactionName: 'foo',
```

Comment:
should we also add another test that it _does_ take the root span name for transaction events?

Reply:
done :) 
(Just a bit painful because the event processor expects a `Span` class instance but with some ts-ignores it all works out 😅 )